### PR TITLE
233 update fetch users logic

### PR DESCRIPTION
**Pull request by MatyashDen**

### CODE REVIEW

**Review by bogdan-evtushenko — DISMISSED** [review]
Lgtm, but need to fix a bug with updating users

File: common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/users/redux/requests/UsersRequests.kt [hunk 1]
```diff
@@ -83,7 +83,7 @@ class UsersRequests {
 
         override suspend fun execute() {
             when (val response = backendRepository.fetchUsers(handle, isAllRatingChangesNeeded = false)) {
-                is Response.Success -> response.result.firstOrNull()?.let { user -> addUser(user) }
+                is Response.Success -> response.result.firstOrNull()?.let { user -> addUser(user) } ?: store.dispatch(Failure(null.toMessage()))
```
Comment (bogdan-evtushenko) [inline common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/users/redux/requests/UsersRequests.kt:86]: I think it's impossible to reach such situation.

**Review by bogdan-evtushenko — APPROVED** [review]
What about omitting flicker of UserItemCeil after returning from UserActivity by writing smart hash function on UserItem (which isn't considering the all rating changes but only last one)?

File: common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/users/redux/requests/UsersRequests.kt (hunk 1 above)
Comment (MatyashDen) [reply common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/users/redux/requests/UsersRequests.kt:86]: when user have russian name

Comment (bogdan-evtushenko) [reply common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/users/redux/requests/UsersRequests.kt:86]: ok

**Pull request merged**